File: src/homework05/ui/ConsoleView.java
package homework05.ui;

import homework05.mvp.View;

import java.util.Scanner;

public class ConsoleView implements View {
    Scanner iScanner;

    public ConsoleView() {
        iScanner = new Scanner(System.in);
    }

    @Override
    public void showString(String s) {
        System.out.print(s);
    }

    @Override
    public void inputPrompt(String field) {
        showString(String.format("Input %s: ", field));
    }

    @Override
    public void showMessage(String s) {
        showString(String.format("%s%n%n", s));
    }

    @Override
    public void showMenu(String s) {
        showString(String.format("%s%n", s));
    }

    @Override
    public Integer getIntegerFieldFromInput() {
        String idString = iScanner.nextLine();

        try {
            return Integer.parseInt(idString);
        } catch (Exception e) {
            return -1;
        }
    }

    @Override
    public String getStringFieldFromInput(String field) {
        inputPrompt(field);
        return iScanner.nextLine();
    }

}

File: src/homework07/phonebook/ui/console_view.py
from phonebook.mvp.view import View


class ConsoleView(View):

    def show_string(self, text: str) -> None:
        print(text, end='')

    def show_message(self, text: str) -> None:
        print(text, end='\n\n')

    def show_menu(self, text: str) -> None:
        self.show_string(f'\n{text}')

    def input_prompt(self, field: str) -> None:
        self.show_string(f'Input {field}: ')

    def get_int_from_input(self) -> int:
        try:
            return int(input())
        except ValueError:
            return -1

    def get_string_field_from_input(self, field: str) -> str:
        self.input_prompt(field)
        return input()
